internal/controller/rotators/aws_oidc_rotator.go

```diff
@@ -0,0 +1,162 @@
+package backendauthrotators
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// AWSOIDCRotator implements the Rotator interface for AWS OIDC token exchange.
+// It manages the lifecycle of temporary AWS credentials obtained through OIDC token
+// exchange with AWS STS.
+type AWSOIDCRotator struct {
+	// client is used for Kubernetes API operations
+	client client.Client
+	// kube provides additional Kubernetes API capabilities
+	kube kubernetes.Interface
+	// logger is used for structured logging
+	logger logr.Logger
+	// stsOps provides AWS STS operations interface
+	stsOps STSOperations
+	// backendSecurityPolicyName provides name of backend security policy
+	backendSecurityPolicyName string
+	// backendSecurityPolicyNamespace provides namespace of backend security policy
+	backendSecurityPolicyNamespace string
+	// preRotationWindow specifies how long before expiry to rotate
+	preRotationWindow time.Duration
+}
+
+// NewAWSOIDCRotator creates a new AWS OIDC rotator with the specified configuration.
+// It initializes the AWS STS client and sets up the rotation channels.
+func NewAWSOIDCRotator(
+	client client.Client,
+	kube kubernetes.Interface,
+	logger logr.Logger,
+	backendSecurityPolicyNamespace string,
+	backendSecurityPolicyName string,
+	preRotationWindow time.Duration,
+	region string,
+) (*AWSOIDCRotator, error) {
+	cfg, err := defaultAWSConfig(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("failed to load AWS config: %w", err)
+	}
+
+	if region != "" {
+		cfg.Region = region
+	}
+
+	if proxyURL := os.Getenv("AI_GATEWY_STS_PROXY_URL"); proxyURL != "" {
+		cfg.HTTPClient = &http.Client{
+			Transport: &http.Transport{
+				Proxy: func(*http.Request) (*url.URL, error) {
+					return url.Parse(proxyURL)
+				},
+			},
+		}
+	}
+
+	stsClient := NewSTSClient(cfg)
+
+	return &AWSOIDCRotator{
+		client:                         client,
+		kube:                           kube,
+		logger:                         logger,
+		stsOps:                         stsClient,
+		backendSecurityPolicyNamespace: backendSecurityPolicyNamespace,
+		backendSecurityPolicyName:      backendSecurityPolicyName,
+		preRotationWindow:              preRotationWindow,
+	}, nil
+}
+
+// SetSTSOperations sets the STS operations implementation - primarily used for testing
+func (r *AWSOIDCRotator) SetSTSOperations(ops STSOperations) {
+	r.stsOps = ops
+}
+
+func (r *AWSOIDCRotator) IsExpired() bool {
```

Comment:
any exported function needs comment since whenever anyone sees this, they would wonder if this is the implementation of an interface or not. Even when the function name is self-explanatory, I would definitely prefer a doc comment what this is  


this applies to here and elsewhere